torchmd/parameters.py

```diff
@@ -114,10 +113,14 @@ def build_parameters(self, ff, mol, terms):
 
         self.mapped_atom_types = torch.tensor(indexes)
         self.charges = torch.tensor(mol.charge.astype(np.float64))
-        if mol.masses is not None and isinstance(mol, npzMolecule):
+        if mol.masses is not None:
             self.masses = torch.tensor(mol.masses).to(torch.float32)[:, None]
-        else:
+        elif np.all(mol.atomtype != "") and ff.prm is not None:
             self.masses = self.make_masses(ff, mol.atomtype)
+        else:
+            raise RuntimeError(
+                "No masses or atomtypes defined in the Molecule. Please define either in the input molecule so that we can get the atom masses. Use mol.masses = [periodictable[el].mass for el in mol.element]"
```

Comment:
"No masses or atomtypes defined in the Molecule. If you are using MM force-field parameters please read the *.prmtop file with the Molecule class so that atomtypes are set correctly. If you are not using MM force-fields set the atom masses as follows: from moleculekit.periodictable import periodictable; mol.masses = [periodictable[el].mass for el in mol.element]"